packages/babylon/src/plugins/estree.js

```diff
@@ -128,11 +128,10 @@ export default (superClass: Class<Parser>): Class<Parser> =>
       }
     }
 
-    isStrictBody(
-      node: { body: N.BlockStatement },
-      isExpression: ?boolean,
-    ): boolean {
-      if (!isExpression && node.body.body.length > 0) {
+    isStrictBody(node: { body: N.BlockStatement }): boolean {
+      const isBlockStatement = node.body.type === "BlockStatement";
+
+      if (isBlockStatement && node.body.body.length > 0) {
         for (const directive of node.body.body) {
```

Comment:
As the flag is part of the estree spec, the estree plugin needs to ensure that this field stays set.